### PR TITLE
Algorand transaction api implemenation

### PR DESCRIPTION
**Pull request by barjonr**

### CODE REVIEW

File: ledger-core-algorand/inc/algorand/model/transactions/AlgorandAsset.hpp [hunk 1]
```diff
@@ -49,14 +49,14 @@ namespace model {
         static AssetConfigTxnFields reconfigure(AssetParams assetParams, uint64_t assetId);
         static AssetConfigTxnFields destroy(uint64_t assetId);
 
-        AssetConfigTxnFields() {}
+        AssetConfigTxnFields() = default;
```
Comment (haammar-ledger) [inline ledger-core-algorand/inc/algorand/model/transactions/AlgorandAsset.hpp:52]: :+1: thanks

File: ledger-core-algorand/inc/algorand/model/transactions/AlgorandAsset.hpp [hunk 2]
```diff
@@ -49,14 +49,14 @@ namespace model {
...
 
     private:
         AssetConfigTxnFields(Option<AssetParams> assetParams, Option<uint64_t> assetId);
 
     public:
-        Option<uint64_t> assetId;
         Option<AssetParams> assetParams;
+        Option<uint64_t> assetId;
```
Comment (haammar-ledger) [inline ledger-core-algorand/inc/algorand/model/transactions/AlgorandAsset.hpp:59]: Why did you move assetId everywhere? ^^

File: ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.hpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        void setAssetFreezeInfo(const api::AlgorandAssetFreezeInfo& info) override;
+        api::AlgorandAssetFreezeInfo getAssetFreezeInfo() const override;
+        std::vector<uint8_t> serialize() const override;
+        void setSignature(const std::vector<uint8_t>& signature) override;
+
+    private:
+        model::SignedTransaction txn;
+    };
+
+    std::shared_ptr<api::AlgorandTransaction> makeAlgorandTransaction(model::Transaction txn);
```
Comment (haammar-ledger) [inline ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.hpp:80]: Shouldn't this be a `static` method?

File: ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+namespace core {
+namespace algorand {
+
+    AlgorandTransactionImpl::AlgorandTransactionImpl(model::SignedTransaction txn)
+        : txn(std::move(txn))
+    {}
+
+    std::string AlgorandTransactionImpl::getId() const
+    {
+        return txn.txn.header.id.getValueOr("0");
```
Comment (haammar-ledger) [inline ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp:46]: Why `0` and not an empty string?

File: ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+    }
+
+    std::string AlgorandTransactionImpl::getType() const
+    {
+        return txn.getType();
+    }
+
+    std::string AlgorandTransactionImpl::getSender() const
+    {
+        return txn.txn.header.sender.getAddress();
```
Comment (haammar-ledger) [inline ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp:56]: `txn.txn` looks so weird :smile: I would name the first one `stxn` maybe?

File: ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+    {
+        txn.txn.header.note = { std::begin(note), std::end(note) };
+    }
+
+    void AlgorandTransactionImpl::setPaymentInfo(const api::AlgorandPaymentInfo& info)
+    {
+        model::PaymentTxnFields details;
+        details.amount = std::stoi(info.amount);
+
+        if (info.closeAddress)
```
Comment (haammar-ledger) [inline ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp:97]: Please put brackets for this `if`, it's safer (also in other places in the file).

File: ledger-core-algorand/inc/algorand/model/transactions/AlgorandAsset.hpp (hunk 2 above)
Comment (barjonr) [reply ledger-core-algorand/inc/algorand/model/transactions/AlgorandAsset.hpp:59]: It does not really matter, it was just to have fields in the same order in the class and the constructors haha

File: ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp (hunk 4 above)
Comment (barjonr) [reply ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp:46]: I did not really think about it, but you are right, an empty string would be much better

File: ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp (hunk 5 above)
Comment (barjonr) [reply ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp:56]: I will change that

File: ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp (hunk 6 above)
Comment (barjonr) [reply ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.cpp:97]: 👍

File: ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.hpp (hunk 3 above)
Comment (barjonr) [reply ledger-core-algorand/src/algorand/transactions/api_impl/AlgorandTransactionImpl.hpp:80]: You are right, I will change the name to makeTransaction then
